go/vt/discovery/tablet_health_check.go

```diff
@@ -272,8 +271,20 @@ func (thc *tabletHealthCheck) checkConn(hc *HealthCheckImpl) {
 			}
 		}()
 
+		// Limit the number of healthcheck connections opened in parallel to avoid high OS-thread
+		// usage due to blocking networking syscalls (eg: DNS lookups, TCP connection opens,
+		// etc). Without this limit it is possible for vtgates watching >10k tablets to hit
+		// the panic: 'runtime: program exceeds 10000-thread limit'.
+		var releaseSemOnce sync.Once
```

Comment:
I really don't like using a `sync.Once` or the callback to know we connected, but this is the best I've come up considering all the abstraction

Reply:
Removed